scope.go

```diff
@@ -478,10 +474,12 @@ func (s *scope) Snapshot() Snapshot {
 }
 
 func (s *scope) Close() error {
-	s.status.Lock()
-	s.status.closed = true
-	close(s.status.quit)
-	s.status.Unlock()
+	if atomic.CompareAndSwapInt32(&s.status.closed, 0, 1) {
+		// prevent any more report loops via tick before forcing report
+		close(s.status.quit)
```

Comment:
Do we have a potential race here if reportLoopRun is triggered after line 477 with the tick, then when you come into here, there's actually 2 reportLoopRun functions running and you only block on one of them completing (ie the other one can still be running when you return from this function).

This may be especially true if you close the baseReporter right afterwards, ie some metrics are not yet flushed.